### ISSUE TITLE
[Question] Detailed status semantic structure does not match regular status structure

### ISSUE THREAD

**Issue by trwnh** [tootsuite/mastodon#8551]
I've been having some issues in developing my userstyle pertaining to detailed statuses -- especially because the component structure differs quite a bit from the structure for a normal status.

## Currently, a status looks something like this:

div tabindex="-1"
- div.focusable **.status__wrapper** (1)
  - div.status
    - div.status **__info** (2)
    - div.status__content
    - div.status **__action-bar** (3)

## A detailed status looks like this:

div tabindex="-1"
- div.focusable
  - div.detailed-status
    - div.detailed-status **__display-name** (2)
    - div.status__content
    - div.detailed-status **__meta** (2)
  - div.detailed-status **__action-bar** (3)

## The differences:

1) The .focusable has no wrapper class in detailed statuses. **I submitted PR #8547** to apply `.detailed-status__wrapper` to the `.focusable` instead of the further-up `.scrollable` (which wraps the entire scrollable area and not just the detailed status).

2) The regular status has `__info` and the detailed status has `__display-name`. **This is actually fine** -- regular statuses nest both the display-name and the timestamp within the `__info` class, and detailed statuses instead have a `__meta` class that includes the full time as well as fave/boost counts.

3) The corresponding `__action-bar` is inside `.status`, but outside `.detailed-status`. 

Since (1) already has a PR and (2) isn't an issue, that just leaves (3). From what I could find, the action bar is being generated in different .js files -- status_action_bar.js for statuses and action_bar.js for detailed statuses. From what I understand of the code, there are minor differences between the two: ImmutablePureComponent vs. PureComponent, replyAll for threads vs. just reply, and some different CSS classes in the return statement. Okay, so far so good...

## Request for clarification

Now, my question is, is there a reason that the action bar is at different levels of the hierarchy? The only reason I can see is that maybe it's so that the detailed-status__action-bar is unaffected by the padding of the detailed-status:

![image](https://user-images.githubusercontent.com/10606431/44942897-74fe2800-ad81-11e8-8cad-1999bb3bd70c.png)

Although, even if the action bar is placed inside the detailed-status, the padding can be adjusted:

![image](https://user-images.githubusercontent.com/10606431/44943000-93652300-ad83-11e8-9c92-28f9ad1d8306.png)

The relevant code is here, I think: https://github.com/tootsuite/mastodon/blob/800325f452200902995602ff441e2f84f684e90a/app/javascript/mastodon/features/status/index.js#L422-L445

Is that indeed the reason, or are there any different reasons I'm overlooking? Is this actually unintended, and would a follow-up PR be appropriate to standardize the structure between the two?

## tl;dr

Basically wondering about:

a) differences between `status_action_bar.js` vs `action_bar.js`
b) action bar hierarchies -- `status_action_bar.js` within the status, but `action_bar.js` within detailed status wrapper (currently .focusable but missing an actual wrapper class which #8547 adds)

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on [2.5.0rc1](https://github.com/tootsuite/mastodon/releases) and not on `master`

**Comment by trwnh** [comment]
@Gargron Could I get a response re: this?

**Comment by Gargron** [comment]
@trwnh There isn't really a reason for either structure, it's dictated by visuals. As long as aria labels are up to par, I don't see a reason to change it around?

**Comment by trwnh** [comment]
@Gargron OK, if it's a "this is what we did and it works so there's no reason to fix it", then that's fine. I was mainly wondering about the redundancy between having two different action bar components that look like they generate mostly the same code. PR #8547 addresses my themeability concern since the .focusable actually wraps both the detailed status and also its action bar -- it's just missing a wrapper class.

This can either be closed as a "fixed by #8547" (which fixes point 1) or "no reason to fix" (regarding point 3), or kept open if point 3 is ever intended to be simplified at some future time. I can't figure out a functional difference between the two action bars JS files, anyway.

**Comment by nightpool** [comment]
i think we *should* simplify point 3 at some point in the future, but there's no reason this needs to be a github issue, it could have been just as easily asked over discord :P

**Comment by trwnh** [comment]
@nightpool One slight issue with that: I am not in whichever Discord this is! A GitHub issue seems like a perfectly reasonable place to ask about code semantics. 😆
